interfaces/machine_parameters/ring_section.py

```diff
@@ -177,98 +165,134 @@ def __init__(self, section_length, alpha_0,
+                self.length_design)
+        else:
+            if not isinstance(orbit_bump, ring_programs.orbit_length_program):
+                orbit_bump = ring_programs.orbit_length_program(orbit_bump)
+            if orbit_bump.timebase == 'by_time':
+                orbit_bump[:, 1, :] += self.length_design
+            else:
+                orbit_bump += self.length_design
+            self._check_and_set_alpha_and_orbit(orbit_bump)
+
         # Setting the linear momentum compaction factor
         # Checking that the synchronous data and the momentum compaction
         # have the same length if defined turn-by-turn, raise a warning if one
         # is defined by turn and the other time based
-        self.alpha_order = 0
-        if not isinstance(alpha_0, dTypes.momentum_compaction):
-            alpha_0 = dTypes.momentum_compaction(alpha_0, order=0)
+        self.alpha_orders = [0]
+        alpha_order_max = 0
+        if not isinstance(alpha_0, ring_programs.momentum_compaction):
+            alpha_0 = ring_programs.momentum_compaction(alpha_0, order=0)
         else:
             if alpha_0.order != 0:
                 raise excpt.InputError(
                     "The order of the datatype passed as keyword " +
                     "argument alpha_%s do not match" % (0))
 
-        self._check_and_set_momentum_compaction(alpha_0, 0)
+        self._check_and_set_alpha_and_orbit(alpha_0, 0)
 
         # Treating non-linear momentum compaction factor if declared
         # Listing all the declared alpha first
         alpha_n = {1: alpha_1, 2: alpha_2}
 
         if alpha_1 is not None:
-            self.alpha_order = 1
+            alpha_order_max = 1
         if alpha_2 is not None:
-            self.alpha_order = 2
+            alpha_order_max = 2
 
         for argument in kwargs:
             if 'alpha' in argument:
                 try:
                     order = int(argument.split('_')[-1])
-                    self.alpha_order = np.max([self.alpha_order, order])
+                    alpha_order_max = np.max([alpha_order_max, order])
                     alpha_n[order] = kwargs[argument]
                 except Exception:
                     raise excpt.InputError(
-                        'The keyword argument '+argument+' was interpreted ' +
-                        'as non-linear momentum compaction factor. ' +
+                        'The keyword argument ' + argument + ' was ' +
+                        'interpreted as non-linear momentum compaction ' +
+                        'factor. ' +
                         'The correct syntax is alpha_n.')
 
         # Setting all the valid non-linear alpha and replacing
         # undeclared orders with zeros
-        self.alpha_orders_defined = [0]
-        for order in range(1, self.alpha_order+1):
+        for order in range(1, alpha_order_max + 1):
             alpha = alpha_n.pop(order, None)
 
             if alpha is None:
                 alpha = 0
```

Comment:
```suggestion
            alpha = alpha_n.pop(order, 0)
```

Reply:
This if statement is placeholder, we can replace `alpha = 0` with `continue` to avoid filling undeclared alpha with 0 values

(1M dollar question, what would `alpha = alpha_n.pop(order, continue)` do?)

Reply:
Ah ok, in that case I'd do:
```python
try:
    alpha = alpha_n.pop(order)
except KeyError:
    continue
```

Reply:
Included